test/e2e/predicates/predicates_test.go

```diff
@@ -410,31 +396,25 @@ var _ = Describe("Predicates", func() {
 			NodeSelector: map[string]string{labelKey: labelValue},
 		}
 
-		var success bool
 		initPod, podErr := k8s.InitTestPod(conf)
 		Ω(podErr).NotTo(HaveOccurred())
-		success, err = kClient.WaitForSchedulerAfterAction(kClient.CreateTestPodAction(initPod, ns), ns, podNameNoTolerations, false)
+		_, err := kClient.CreatePod(initPod, ns)
 		Ω(err).NotTo(HaveOccurred())
-		Ω(success).Should(BeTrue())
 
 		By(fmt.Sprintf("Verify pod:%s is in pending state", podNameNoTolerations))
 		err = kClient.WaitForPodPending(ns, podNameNoTolerations, time.Duration(60)*time.Second)
 		Ω(err).NotTo(HaveOccurred())
 
-		By("Verify the Yunikorn events: failed scheduling")
-		var events *v1.EventList
-		events, err = kClient.GetEvents(ns)
+		By("Verify the YuniKorn request failed scheduling")
+
+		err = restClient.WaitForAllocationLog("default", "root."+ns, initPod.ObjectMeta.Labels["applicationId"], podNameNoTolerations, 60)
 		Ω(err).NotTo(HaveOccurred())
-		Ω(len(events.Items)).NotTo(BeZero(), "Events cant be empty")
-		for _, event := range events.Items {
-			framework.Logf("Event source is : %s", event.Source.Component)
-			if event.Source.Component == configmanager.SchedulerName && event.Type == WARNING && !strings.Contains(event.Message, "node affinity") {
-				framework.Logf("Yunikorn Event Reason: %s", event.Reason)
-				Ω(event.Reason).Should(Equal("FailedScheduling"), "Event reason mismatch")
-				framework.Logf("Yunikorn Event Message: %s", event.Message)
-				Ω(event.Message).Should(MatchRegexp("predicate is not satisfied.*taint.*"), "Event message mismatch")
-			}
-		}
+		log, err := restClient.GetAllocationLog("default", "root."+ns, initPod.ObjectMeta.Labels["applicationId"], podNameNoTolerations)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(len(log)).NotTo(BeZero(), "Log can't be empty")
+
+		Ω(log).Should(ContainElement(HaveKeyWithValue("message", MatchRegexp(".*taint.*"))),
```

Comment:
Please see my earlier comment. Just depending only on ".*taint.*" may mislead us sometimes.

Reply:
Per previous comment, we specifically keep these generic to make supporting a wider range of K8s releases earlier. What we match here is the same as the previous e2e tests, which have been working fine.